test/IDE/coloring.swift

```diff
@@ -592,7 +670,8 @@ func keywordAsLabel4(_: Int) {}
 func keywordAsLabel5(_: Int, for: Int) {}
 // CHECK: <kw>func</kw> keywordAsLabel5(<kw>_</kw>: <type>Int</type>, for: <type>Int</type>) {}
 func keywordAsLabel6(if let: Int) {}
-// CHECK: <kw>func</kw> keywordAsLabel6(if <kw>let</kw>: <type>Int</type>) {}
+// CHECK-OLD: <kw>func</kw> keywordAsLabel6(if <kw>let</kw>: <type>Int</type>) {}
+// CHECK-NEW: <kw>func</kw> keywordAsLabel6(<kw>if</kw> <kw>let</kw>: <type>Int</type>) {}
```

Comment:
@rintaro Following up on your comment in #16636: This is however the way the syntax tree is currently built. We could look into optimising the syntax tree creation for this, but to be honest, I don't have strong enough feelings about how this gets coloured to put a lot of effort into it right now. Plus it's clearly marked as a divergence, so when we come back and try to make the new colouring behave exactly the same as the old one, it will come back up.

Reply:
Ah, now I understand why this is `<kw>`, because `let` keyword is not `canBeArgumentLabel()`. this `func` decl is currently a unknown decl.
I don't think that is what we wanted to test. Could you make this `func keywordAsLabel6(if func:` for example?

Reply:
Updated the test case